mapie/metrics.py

```diff
@@ -307,3 +308,397 @@ def top_label_ece(
         )
     ece /= len(labels)
     return ece
+
+
+def regression_coverage_score_v2(
+    y_true: NDArray,
+    y_intervals: NDArray,
+) -> NDArray:
+    """
+    Effective coverage score obtained by the prediction intervals.
+
+    The effective coverage is obtained by estimating the fraction
+    of true labels that lie within the prediction intervals.
+
+    Parameters
+    ----------
+    y_true : NDArray of shape (n_samples,)
+        True labels.
+    y_intervals : NDArray of shape (n_samples, 2, n_alpha)
+        Lower and upper bound of prediction intervals
+        with different alpha risks.
+
+    Returns
+    -------
+    NDArray of shape (n_alpha,)
+        Effective coverage obtained by the prediction intervals.
+    """
+    coverages = np.mean(
+        np.logical_and(
+            np.less_equal(y_intervals[:, 0, :], y_true),
+            np.greater_equal(y_intervals[:, 1, :], y_true)
+        ),
+        axis=0
+    )
+    return coverages
+
+
+def classification_coverage_score_v2(
+    y_true: NDArray,
+    y_pred_set: NDArray
+) -> NDArray:
+    """
+    Effective coverage score obtained by the prediction sets.
+
+    The effective coverage is obtained by estimating the fraction
+    of true labels that lie within the prediction sets.
```

Comment:
Add a brief comment to explain why it is different from the previous version (classification_coverage_score_v2 uses directly the output of predict method = sets)

Reply:
Why is it resolved?